src/Encamina.Enmarcha.SemanticKernel/MemoryManager.cs

```diff
@@ -26,12 +27,22 @@ public class MemoryManager : IMemoryManager
     /// Initializes a new instance of the <see cref="MemoryManager"/> class.
     /// </summary>
     /// <param name="memoryStore">A valid instance of a <see cref="IMemoryStore"/> to manage.</param>
-    public MemoryManager(IMemoryStore memoryStore, ILogger<MemoryManager> logger)
+    /// <param name="memoryStorageEventHandlerDelegate">A delegate to handle events from <see cref="MemoryStorageEvent"/>.</param>
+    /// <param name="logger">Log service.</param>
+    public MemoryManager(IMemoryStore memoryStore, ILogger<MemoryManager> logger, Action<string, MemoryStorageEventArgs> memoryStorageEventHandlerDelegate)
     {
-        this.logger = logger;
         MemoryStore = memoryStore;
+        this.logger = logger;
+
+        if (memoryStorageEventHandlerDelegate is not null)
+        {
+            MemoryStorageEvent += (sender, args) => memoryStorageEventHandlerDelegate?.Invoke(sender?.ToString() ?? MemoryStorageEventEnum.Undefined.GetEnumDescription(), args);
```

Comment:
Idem. It should be `nameof(MemoryStorageEventEnum.Undefined)`.

Reply:
You are right, it is better as you say